### PR TITLE
It change the logic of ability to manage a move.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://dsdmoj.atlassian.net/browse/P4-1899

### What?

I have altered the ability to manage a move

### Why?
In future we intend to support locations having multiple suppliers so we need to adapt our authorisation model.

### CODE REVIEW

**Comment by willfish** [comment]
This is blocked until we backfill moves/do a bit of verification on the state of production data. On staging/preprod, for example, this will render some moves unmanageable by anyone but the frontend (see [here](https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/713))

File: spec/requests/api/moves_controller_create_v2_spec.rb [hunk 1]
```diff
@@ -35,7 +35,8 @@
     end
 
     let(:profile) { create(:profile) }
-    let(:from_location) { create :location, suppliers: [supplier] }
+    let(:another_supplier) { create(:supplier) }
```
Comment (willfish) [inline spec/requests/api/moves_controller_create_v2_spec.rb:38]: You actually don't need any changes to this file :).

File: spec/requests/api/moves_controller_role_access_rswag_spec.rb [hunk 2]
```diff
@@ -23,15 +23,18 @@
...
-    let!(:pentonville) { create :location, :with_moves, suppliers: [pentonville_supplier] }
-    let!(:birmingham) do
-      create :location,
-             :with_moves,
-             key: 'hmp_birmingham',
-             title: 'HMP Birmingham',
-             nomis_agency_id: 'BMI',
-             suppliers: [birmingham_supplier]
-    end
+    # let!(:pentonville) { create :location, :with_moves, suppliers: [pentonville_supplier] }
```
Comment (willfish) [inline spec/requests/api/moves_controller_role_access_rswag_spec.rb:26]: These comments can be removed :).

File: spec/requests/api/moves_controller_role_access_rswag_spec.rb [hunk 3]
```diff
@@ -159,34 +162,13 @@
         expect(response_json).to eq resource_to_json
       end
     end
-
-    context 'when supplier doesn\'t have rights to write the resource' do
```
Comment (willfish) [inline spec/requests/api/moves_controller_role_access_rswag_spec.rb:163]: 👍

File: spec/requests/api/moves_controller_role_access_rswag_spec.rb [hunk 4]
```diff
@@ -23,15 +23,18 @@
...
-      create :location,
-             :with_moves,
-             key: 'hmp_birmingham',
-             title: 'HMP Birmingham',
-             nomis_agency_id: 'BMI',
-             suppliers: [birmingham_supplier]
-    end
+    # let!(:pentonville) { create :location, :with_moves, suppliers: [pentonville_supplier] }
+
+    let!(:moves) { create_list(:move, 2, supplier: pentonville_supplier) }
```
Comment (willfish) [inline spec/requests/api/moves_controller_role_access_rswag_spec.rb:28]: We need to make sure we're creating moves for the birmingham supplier:

```ruby
    let!(:filtered_out_moves) { create_list(:move, 2, supplier: birmingham_supplier) }
```

Although the name birmingham_supplier implies 1-2-1 relationship between a location and a supplier which of course isn't correct.

File: spec/requests/api/moves_controller_role_access_rswag_spec.rb [hunk 5]
```diff
@@ -44,7 +47,7 @@
 
       it 'returns only moves belonging to suppliers' do
         response_ids = response_json['data'].map { |move| move['id'] }.sort
-        data_ids = pentonville.moves_from.pluck(:id).sort
```
Comment (willfish) [inline spec/requests/api/moves_controller_role_access_rswag_spec.rb:47]: Sorting by id here is worrying, tbh. We should really define standard orders for collections of our entities and optimise by ordered indexes 🤔 for the default sort order. Another PR. If we don't care about the order maybe we can use the `contain_exactly` matcher and remove the `sort` on line 47

File: spec/requests/api/moves_controller_create_v2_spec.rb (hunk 1 above)
Comment (alexdesi) [reply spec/requests/api/moves_controller_create_v2_spec.rb:38]: we discussed this, and it makes sense to keep it

**Comment by willfish** [comment]
```
 Running tests in:
 - Chrome 83.0.4103.116 / macOS 10.15.5

 Cancel allocation
 ✓ Cancel allocation

 New PMU allocation
 ✓ Create allocation and verify the result
 ✓ Check validation errors on allocation details page

 New move from Police Custody to Court
 ✓ With unfound person
 ✓ With existing person

 New move from Police Custody to Prison (recall)
 ✓ With a new person

 Cancel move from Police Custody
 ✓ Reason - `Made in error`
 ✓ Reason - `Supplier declined to move this person`
 ✓ Reason - `Another reason`

 New move from Prison to Court
 ✓ With unfound person
 ✓ With existing person

 New move from Prison to Prison
 ✓ With existing person

 Download moves
 ✓ As Police user
 ✓ As Prison user
 ✓ As STC user
 ✓ As Supplier user

 Smoke tests
 ✓ As Police user
 ✓ As STC user
 ✓ As Prison user
 ✓ As Supplier user
 ✓ As OCA user


 21 passed (5m 59s)
```

**Pull request merged**